tests/components/network_sediment_transporter/test_bed_initializer.py

```diff
@@ -0,0 +1,272 @@
+"""
+Tests written by Jeff Keck and Allison Pfeiffer
+"""
+
+import numpy as np
+import pytest
+
+from landlab.components import (
+    BedParcelInitializerDepth,
+    BedParcelInitializerArea,
+    BedParcelInitializerUserD50,
+    BedParcelInitializerDischarge,
+)
+
+from landlab.components.network_sediment_transporter.bed_parcel_initializers import (
+    _determine_approx_parcel_volume,
+    calc_total_parcel_volume,
+    calc_d50_discharge,
+    calc_d50_depth,
+    calc_d50_dArea_scaling,
+)
+
+
+# Basic test: that you can call the four initializers
+def test_call_area_BPI(example_nmg2):
+    initialize_parcels = BedParcelInitializerArea(
+        example_nmg2, drainage_area_coefficient=0.18, drainage_area_exponent=-0.12
+    )
+    _ = initialize_parcels()
+
+
+def test_call_discharge_BPI(example_nmg2):
+    initialize_parcels = BedParcelInitializerDischarge(
+        example_nmg2, discharge_at_link=np.full(example_nmg2.number_of_links, 10.0)
+    )
+    _ = initialize_parcels()
+
+
+def test_call_depth_BPI(example_nmg2):
+    depth_at_link = np.full(example_nmg2.number_of_links, 10.0)  # m
+    initialize_parcels = BedParcelInitializerDepth(
+        example_nmg2,
+        flow_depth_at_link=depth_at_link,
+    )
+    _ = initialize_parcels()
+
+
+def test_call_userD50_BPI(example_nmg2):
+    d50_at_link = np.full(example_nmg2.number_of_links, 0.08)  # m
+    initialize_parcels = BedParcelInitializerUserD50(
+        example_nmg2,
+        user_d50=d50_at_link,
+    )
+    _ = initialize_parcels()
+
+
+# %% Tests of BPI base class helper functions
+
+
+def test_calc_total_parcel_volume():
+    vole = 12.5
+    vol = calc_total_parcel_volume(5, 5, 0.5)
+    np.testing.assert_allclose(vol, vole, rtol=1e-4)
+
+
+def test_det_approx_parcel_volume():
+    total_parcel_volume_at_link = 10
+    median_number_of_starting_parcels = 50
+    pvole = 0.2
+    pvol = _determine_approx_parcel_volume(
+        total_parcel_volume_at_link, median_number_of_starting_parcels
+    )
+    np.testing.assert_allclose(pvol, pvole, rtol=1e-4)
+
+
+# %% Test to make sure we're getting correct correct d50 correct_values
+
+
+def test_calc_D50_discharge():
+    """test calc D50 grain size give correct values"""
+    width = 20
+    slope = 0.01
+    discharge = 100
+    mannings_n = 0.05
+    gravity = 9.81
+    rho_water = 1000
+    rho_sediment = 3000
+    tau_c_50 = 0.05
+
+    D50e = 0.173286
```

Comment:
How do you know this is the expected value? Did you hand calculate it? If so, that's ok but if you just ran *calc_d50_discharge* and then copied the output, then I would rather see a better test here.

Reply:
Hand calculated from Shields stress equation.